pkg/process/checks/discovery_check_test.go

```diff
@@ -0,0 +1,34 @@
+package checks
+
+import (
+	"testing"
+
+	model "github.com/DataDog/agent-payload/process"
+	"github.com/DataDog/datadog-agent/pkg/process/config"
+)
+
+func TestProcessDiscoveryCheck(t *testing.T) {
+	cfg := &config.AgentConfig{MaxPerMessage: 10}
+	ProcessDiscovery.Init(cfg, &model.SystemInfo{
+		Cpus:        []*model.CPUInfo{{Number: 0}},
+		TotalMemory: 0,
+	})
+
+	// Test check runs without error
+	result, err := ProcessDiscovery.Run(cfg, 0)
+	if err != nil {
+		t.Error(err)
+	}
```

Comment:
nit: the following my be more succinct and will skip running anything below it.
 `require.NoError(t, err)`